caravel/templates/listings/inline.html

```diff
@@ -1,27 +1,52 @@
-<div class="col-xs-12 col-sm-3 thumbnail-col">
-<div class="inline-thumbnail">
-  <div class="category-tag">
-    <span class="category-text"> Category </span>
+<div class="col-xs-12 col-sm-3 float-to-inline-block listing">
+  <div class="thumbnail">
+    <a href="{{ url_for('show_listing', permalink=listing.permalink,
+                  q=request.args.get('q')) }}">
+        <span class="label label-info" style="background-color:black;opacity:0.8;font-size:13px;position:absolute;margin-top:10px;margin-left:10px;">
+          {{ listing.posting_time|as_duration }} ago</span>
+      {% if listing.thumbnail_url %}
+        <img src="{{ listing.thumbnail_url | public_url }}"/>
+      {% else %}
+        <img src="/static/no-image.png"/>
+      {% endif %}
+    </a>
+    <div class="caption">
+    	<p>
+        <a href="{{ url_for('show_listing', permalink=listing.permalink,
+    	                 q=request.args.get('q')) }}"
+    	       style="color:black;text-decoration:none;font-size:20px">
+    	        {% if listing.title[50:] %}
+                {{ listing.title[:50] }}...
+              {% else %}
+                {{ listing.title }}
+              {% endif %}
+        </a>
+      </p>
+      <p>
+        {% set category_colors = {
+          "apartments":"#a52a2a",
+          "subleases": "#ff1493",
+          "appliances": "#ff8c00",
+          "bikes": "#6495ed",
+          "cars": "#ff0000",
+          "electronics": "#9370db",
+          "employment": "#008000",
+          "furniture": "#ff7f50",
+          "miscelleneous": "#808080",
+          "services": "#202baa",
+          "wanted": "#40e0d0"
+          }
+        %}
+        {% for category in listing.categories %}
+          <a href="{{ url_for('search_listings',
+              q=request.args.get('q', '') + category) }}"
```

Comment:
Change to `.get('q', '') + ' ' + category`; otherwise, if you search for something and then click on a link, bad stuff happens.